libs/api/repository/src/lib/gn4/gn4-repository.ts

```diff
@@ -365,6 +367,47 @@ export class Gn4Repository implements RecordsRepositoryInterface {
     return of(draftCount)
   }
 
+  hasRecordChangedSinceDraft(localRecord: CatalogRecord) {
+    const isUnsaved = this.isRecordNotYetSaved(localRecord.uniqueIdentifier)
+    const hasDraft = this.recordHasDraft(localRecord.uniqueIdentifier)
+
+    if (isUnsaved || !hasDraft) {
+      return of([])
+    }
+
+    return combineLatest([
+      this.getAllDrafts().pipe(
+        map((drafts) => {
+          const matchingRecord = drafts.find(
+            (draft) => draft.uniqueIdentifier === localRecord.uniqueIdentifier
+          )
+          return matchingRecord ? matchingRecord.recordUpdated : null
+        })
+      ),
+      this.getRecord(localRecord.uniqueIdentifier),
+    ]).pipe(
+      map(([draftRecordUpdated, recentRecord]) => {
+        if (recentRecord.recordUpdated > draftRecordUpdated) {
+          return [
+            this.formatDate(recentRecord.recordUpdated),
+            recentRecord.extras?.['ownerInfo'].toString().split('|')[0],
+          ]
```

Comment:
I'd rather return a typed object with a `date` and a `user` here than an array. That would make clearer what info is returned, IMO. Formatting might be done at component level, I think.